sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/Grpc/ServiceBusGrpcSessionEndToEndTests.cs

```diff
@@ -214,6 +301,119 @@ public class ServiceBusBindToSessionMessageAndDefer
             }
         }
 
+        public class ServiceBusBindToSessionMessageAndRenewMessageLock
+        {
+            internal static SettlementService SettlementService { get; set; }
+            public static async Task BindToMessage(
+                [ServiceBusTrigger(FirstQueueNameKey, IsSessionsEnabled = true)] ServiceBusReceivedMessage message, ServiceBusReceiveActions receiveActions)
+            {
+                Assert.AreEqual("foobar", message.Body.ToString());
+                await SettlementService.RenewMessageLock(
+                    new RenewMessageLockRequest
+                    {
+                        Locktoken = message.LockToken,
+                    },
+                    new MockServerCallContext());
+                _waitHandle1.Set();
+            }
+        }
+
+        public class ServiceBusBindToSessionMessageAndSetAndGet
+        {
+            internal static SettlementService SettlementService { get; set; }
+            public static async Task BindToMessage(
+                [ServiceBusTrigger(FirstQueueNameKey, IsSessionsEnabled = true)] ServiceBusReceivedMessage message, ServiceBusReceiveActions receiveActions)
+            {
+                Assert.AreEqual("foobar", message.Body.ToString());
+                await SettlementService.SetSessionState(
+                    new SetSessionStateRequest
+                    {
+                        SessionId = message.SessionId,
+                        SessionState = ByteString.CopyFromUtf8(message.Body.ToString())
+                    },
+                    new MockServerCallContext()
+                 );
+                var test = await SettlementService.GetSessionState(
+                    new GetSessionStateRequest
+                    {
+                        SessionId = message.SessionId,
+                    },
+                    new MockServerCallContext());
+                Assert.IsNotEmpty(test.SessionState);
+                Assert.AreEqual("foobar", message.Body.ToString());
+                _waitHandle1.Set();
+            }
+        }
+
+        public class ServiceBusBindToSessionMessageAndSetAndGetBinaryData
+        {
+            internal static SettlementService SettlementService { get; set; }
+            public static async Task BindToMessage(
+                [ServiceBusTrigger(FirstQueueNameKey, IsSessionsEnabled = true)] ServiceBusReceivedMessage message, ServiceBusReceiveActions receiveActions)
+            {
+                byte[] predefinedData = { 0x48, 0x65 };
+                Assert.AreEqual(predefinedData, message.Body.ToArray());
+                await SettlementService.SetSessionState(
+                    new SetSessionStateRequest
+                    {
+                        SessionId = message.SessionId,
+                        SessionState = ByteString.CopyFromUtf8(message.Body.ToString())
```

Comment:
since the body is binary data, we shouldn't be doing a ToString and instead should rely on some Grpc method that allows creating a Bytestring from binary data directly.

Reply:
Would using `ByteString.CopyFrom(predefinedData)` be ok?

Reply:
Since session state is required to be of type bytes

Reply:
Yes I think so.